dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/support/ClusterUtils.java

```diff
@@ -81,10 +81,16 @@ public static URL mergeUrl(URL remoteUrl, Map<String, String> localMap) {
 
         if (localMap != null && localMap.size() > 0) {
             Map<String, String> copyOfLocalMap = new HashMap<>(localMap);
-            copyOfLocalMap.remove(GROUP_KEY);
+
+            if(map.containsKey(GROUP_KEY)){
+                copyOfLocalMap.remove(GROUP_KEY);
+            }
+            if(map.containsKey(VERSION_KEY)){
+                copyOfLocalMap.remove(VERSION_KEY);
+            }
+

```

Comment:
This change means that if the provider set the group or version, the group or version set by the consumer will be override. I think it's not right, because it may cause wrong invoke.
I think the right solution is that just no remove group and version from localMap.

Reply:
OK,I once think should no remove from localMap,  But I check previous version of ClusterUtils, It always is  "if the provider set the version, the version set by the consumer will be override", I think this logic maybe use in RegistryDirectory.mergeUrl
![image](https://user-images.githubusercontent.com/9245774/62819108-20fbe600-bb83-11e9-9b29-c33e99cae12d.png)